### ISSUE TITLE
twisted.python.runtime.platform should say if the platform supports symlinks

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8292]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8292|
|Type|enhancement|
|Created|2016-04-14 12:53:36Z|
|Branch|https://github.com/twisted/twisted/tree/supportsSymlinks-8292|

Part of [#8025](https://github.com/twisted/twisted/issues/8025), we can't trust Python when we ask it on Python3 on Vista+, as it has symlink support that is technically usable, but only with a special permission setting which regular users don't have.

<details><summary>Searchable metadata</summary>

```
trac-id__8292 8292
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_supportsSymlinks_8292 branches/supportsSymlinks-8292
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1460638416676628 1460638416676628
changetime__1460684031606382 1460684031606382
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [47231]) Branching to supportsSymlinks-8292.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Happy with this too! Please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @hawkowl|
|-|-|

Looks good. Please merge.

Maybe just a minor change. Instead of

```
Check for symlink support.
```

we can have


```
Check for symlink support in Twisted.
```

to make it clear that this is a special support and not the absolute support for Python on that os.

-----

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

(In [47238]) Merge supportsSymlinks-8292: Add symlink support checks to t.p.runtime.platform

Author: hawkowl
Reviewer: adiroiban
Fixes: [#8292](https://github.com/twisted/twisted/issues/8292)
